test/Kestrel.FunctionalTests/TestHelpers/H2SpecCommands.cs

```diff
@@ -0,0 +1,215 @@
+            }
+            if (line.StartsWith("Generic"))
+            {
+                groupName = "generic";
+                return true;
+            }
+            if (line.StartsWith("HPACK"))
+            {
+                groupName = "hpack";
+                return true;
+            }
+            throw new InvalidOperationException("Unrecognized line: " + line);
+        }
+
+        // "8.1.2.1. Pseudo-Header Fields"
+        private static bool IsSectionLine(string line, out string section)
+        {
+            line = line.TrimStart();
+            var firstSpace = line.IndexOf(" ");
+            if (firstSpace < 2) // Minimum: "8. description"
+            {
+                section = string.Empty;
+                return false;
+            }
+
+            // As opposed to test cases that are marked with :
+            if (line[firstSpace - 1] == '.')
+            {
+                section = line.Substring(0, firstSpace - 1); // Drop the trailing dot.
+                return true;
+            }
+
+            section = string.Empty;
+            return false;
+        }
+
+        // "1: Sends a DATA frame"
+        private static bool IsTestLine(string line, out string testNumber, out string description)
+        {
+            line = line.TrimStart();
+            var firstSpace = line.IndexOf(" ");
+            if (firstSpace < 2) // Minimum: "8: description"
+            {
+                testNumber = string.Empty;
+                description = string.Empty;
+                return false;
+            }
+
+            // As opposed to test cases that are marked with :
+            if (line[firstSpace - 1] == ':')
+            {
+                testNumber = line.Substring(0, firstSpace - 1); // Drop the trailing colon.
+                description = line.Substring(firstSpace + 1);
+                return true;
+            }
+
+            testNumber = string.Empty;
+            description = string.Empty;
+            return false;
+        }
+        
+        public static void RunTest(string testId, int port, bool https, ILogger logger)
+        {
+            var tempFile = Path.GetTempPath() + Guid.NewGuid() + ".xml";
+            var processOptions = new ProcessStartInfo()
+            {
+                FileName = GetToolLocation(),
+                RedirectStandardOutput = true,
+                Arguments = $"{testId} -p {port.ToString(CultureInfo.InvariantCulture)} --strict -j {tempFile}"
+                    + (https ? " --tls --insecure" : ""),
+                WindowStyle = ProcessWindowStyle.Hidden,
+                CreateNoWindow = true,
+            };
+
+            using (var process = Process.Start(processOptions))
+            {
+                var data = process.StandardOutput.ReadToEnd();
+                logger.LogDebug(data);
+
+                var results = File.ReadAllText(tempFile);
+                File.Delete(tempFile);
+
+                var xml = new XmlDocument();
```

Comment:
XDocument?

Reply:
Any specific reason?

Reply:
All cool kids use it :)

Reply:
You should know better than to give into peer pressure.